src/module/item/effect/data.ts

```diff
@@ -44,6 +44,7 @@ interface EffectSystemData
     expired: boolean;
     badge: EffectBadge | null;
     remaining: string;
+    looping: boolean | null;
```

Comment:
What sets this to null?

Reply:
And let's save few bytes as call it `loop`

Reply:
And, and, and: this is badge data, so it shouldn't be at the top level.

Reply:
Pushed it down into BadgeCounter as an optional property so it didn't break all the inheritors. At worst it would be `undefined` so I put a boolean conversion in the increase() function, but I can move that to data preparation if that would be cleaner or more efficient.

Reply:
Please do set it during data preparation.

Reply:
Done in latest commit.